internal/impl/sql/processor_sql_raw.go

```diff
@@ -213,70 +260,123 @@ func (s *sqlRawProcessor) ProcessBatch(ctx context.Context, batch service.Messag
 	s.dbMut.RLock()
 	defer s.dbMut.RUnlock()
 
-	var argsExec *service.MessageBatchBloblangExecutor
-	if s.argsMapping != nil {
-		argsExec = batch.BloblangExecutor(s.argsMapping)
+	argsExec := make([]*service.MessageBatchBloblangExecutor, len(s.queries))
+	for i, q := range s.queries {
+		if q.argsMapping != nil {
+			argsExec[i] = batch.BloblangExecutor(q.argsMapping)
+		}
+	}
+	dynQueries := make([]*service.MessageBatchInterpolationExecutor, len(s.queries))
+	for i, q := range s.queries {
+		if q.dynamic != nil {
+			dynQueries[i] = batch.InterpolationExecutor(q.dynamic)
+		}
 	}
 
 	batch = batch.Copy()
+
 	for i, msg := range batch {
-		var args []any
-		if argsExec != nil {
-			resMsg, err := argsExec.Query(i)
+		var tx *sql.Tx
+		var err error
+		if len(s.queries) > 1 {
+			tx, err = s.db.BeginTx(ctx, nil)
 			if err != nil {
-				s.logger.Debugf("Arguments mapping failed: %v", err)
 				msg.SetError(err)
 				continue
 			}
-
-			iargs, err := resMsg.AsStructured()
-			if err != nil {
-				s.logger.Debugf("Mapping returned non-structured result: %v", err)
-				msg.SetError(fmt.Errorf("mapping returned non-structured result: %w", err))
-				continue
+		}
+		argsUpdated := false
+		for j, query := range s.queries {
+			var args []any
+			if argsExec[j] != nil {
+				var resMsg *service.Message
+				if argsUpdated {
+					exec := batch.BloblangExecutor(query.argsMapping)
+					resMsg, err = exec.Query(i)
+				} else {
+					resMsg, err = argsExec[j].Query(i)
+				}
+				if err != nil {
+					s.logger.Debugf("Arguments mapping failed: %v", err)
```

Comment:
Might be worth consolidating these log statements into one outside of the `for` loop

Reply:
Good call done.